docs/learn/smart-contract-developers/create-lsp7-token.md

```diff
@@ -108,55 +152,78 @@ The `privateKey` coming from your UP extension is the private key of the EOA tha
+5. Deploy the contract
 
 ```ts title="scripts/deployUP.ts"
-import hre from 'hardhat';
 import { ethers } from 'hardhat';
 import * as dotenv from 'dotenv';
-import UniversalProfile from '@lukso/lsp-smart-contracts/artifacts/LSP0ERC725Account.json';
+import UniversalProfileArtifact from '@lukso/lsp-smart-contracts/artifacts/LSP0ERC725Account.json';
+import { CustomToken__factory } from '../typechain-types';
 
 // load env vars
 dotenv.config();
-const { UP_ADDR, PRIVATE_KEY } = process.env;
+const { UP_ADDR } = process.env;
 
 async function main() {
-  // setup provider
-  const provider = new ethers.JsonRpcProvider(
-    'https://rpc.testnet.lukso.network',
-  );
-  // setup signer (the browser extension controller)
-  const signer = new ethers.Wallet(PRIVATE_KEY as string, provider);
   // load the associated UP
-  const UP = await ethers.getContractAt('UniversalProfile', UP_ADDR as string);
-  console.log('🔑 EOA: ', signer.address);
-  console.log('🆙 Universal Profile: ', await UP.getAddress());
+  const universalProfile = await ethers.getContractAtFromArtifact(
+    UniversalProfileArtifact,
+    UP_ADDR as string,
+  );
 
   /**
    * Custom LSP7 Token
    */
   console.log('⏳ Deploying the custom Token');
-  const CustomTokenBytecode =
-    hre.artifacts.readArtifactSync('CustomToken').bytecode;
+  const CustomTokenBytecode = CustomToken__factory.bytecode; // bytecode of the custom token without the constructor params
+
+  // custom token constructor params
+  const tokenName = 'My Custom Token';
+  const tokenSymbol = 'MCT';
+  const tokenContractOwnerAddress = (await ethers.getSigners())[0].address;
+  const tokenType = 0; // generic token
+  const isNonDivisible = false;
+
+  // encode constructor params
+  const abiEncoder = new ethers.AbiCoder();
+  const encodedConstructorParams = abiEncoder.encode(
+    ['string', 'string', 'address', 'uint256', 'bool'],
+    [
+      tokenName,
+      tokenSymbol,
+      tokenContractOwnerAddress,
+      tokenType,
+      isNonDivisible,
+    ],
+  );
 
-  // get the address of the contract that will be created
-  const CustomTokenAddress = await UP.connect(signer)
-    .getFunction('execute')
-    .staticCall(1, ethers.ZeroAddress, 0, CustomTokenBytecode);
+  // add the constructor params to the Custom Token bytecode
+  const CustomTokenBytecodeWithConstructor =
+    CustomTokenBytecode + encodedConstructorParams.slice(2);
 
-  // deploy CustomLSP7 as the UP (signed by the browser extension controller)
-  const tx1 = await UP.connect(signer).execute(
+  // get the address of the Custom Token contract that will be created
+  const CustomTokenAddress = await universalProfile.staticCall(
     1,
     ethers.ZeroAddress,
     0,
-    CustomTokenBytecode,
+    CustomTokenBytecodeWithConstructor,
   );
 
-  await tx1.wait();
+  // deploy the Custom Token contract
+  const tx = await universalProfile.execute(
+    1,
+    ethers.ZeroAddress,
+    0,
+    CustomTokenBytecodeWithConstructor,
+  );
```

Comment:
Maybe to add some context. 
Users reading this guide will prob be very now to the whole LUKSO thing so the more way re route to other docs etc the better.

```suggestion
  // deploy the Custom Token contract
  // https://docs.lukso.tech/contracts/contracts/ERC725/#execute
  const tx = await universalProfile.execute(
    1, // Operation type: CREATE
    ethers.ZeroAddress,
    0, // Value is empty
    CustomTokenBytecodeWithConstructor,
  );
```